checker/src/main/java/org/checkerframework/checker/initialization/InitializationStore.java

```diff
@@ -157,6 +156,12 @@ protected CFValue newFieldValueAfterMethodCall(
     protected boolean isDeclaredInitialized(FieldAccess fieldAccess) {
         InitializationParentAnnotatedTypeFactory atypeFactory =
                 (InitializationParentAnnotatedTypeFactory) analysis.getTypeFactory();
+
+        AnnotatedTypeMirror declField = atypeFactory.getAnnotatedType(fieldAccess.getField());
+        if (!declField.hasAnnotation(atypeFactory.INITIALIZED)) {
+            return false;
```

Comment:
This method now only returns true if both the declared field type and the receiver are `Initialized`.
Can you add a test where this is not good enough, i.e. a `NotOnlyInitialized` field accessed through an initialized receiver? Or what is an example that would require the viewpoint adaptation?

Reply:
I don't think there is an example that needs the adaption. This method is used to decide whether the refined init type of a field should be removed from the store after a method call, which only matters if the receiver is not `Initialized`. in the latest commit, I replaced the whole method by a single line, which still works for all test cases.

Reply:
That still doesn't solve the performance issue though.